zendframework/ZendSkeletonApplication
# Prepare for zend-mvc v3

For the v3 release, we will be pulling in a smaller set of components out-of-the-box. These are primarily defined by zend-mvc, but also include:

- ~~zend-config (used by the zend-modulemanager `ConfigListener` by default)~~ (zend-modulemanager now defines the zend-config requirement)
- [zend-skeleton-installer](https://zendframework.github.io/zend-skeleton-installer/) for prompting for optional packages/capabilities.
- [zend-component-installer](https://zendframework.github.io/zend-component-installer), to automate registration of components as modules.

Additionally, we plan to leverage Composer for all autoloading, which means:

- we can register a PSR-4 entry for the `Application` module, and remove one directory level;
- the `Application` module no longer needs to setup autoloading.

For a minimal application, we do not want or need internationalization. As such, this patch also *removes all translations* and calls to translate strings in templates. We can explore re-adding them later, or offering an alternative skeleton application with more default features (including i18n, forms, databases, etc.).

At the time of submission, this patch provides a working skeleton with the current zend-mvc develop branch.

## Testing

If you wish to test:

- Clone the repository locally
- Enter the cloned repository
- Switch to the develop branch: `git checkout -b develop origin/develop`
- Apply the changes: `wget https://patch-diff.githubusercontent.com/raw/zendframework/ZendSkeletonApplication/pull/329.patch && git am 329.patch` (or, if you use the hub command: `git merge https://github.com/zendframework/ZendSkeletonApplication/pull/329`)
- Execute `composer install`

If you want to use vagrant:

- Start it: `vagrant up`
- Run composer within it: `vagrant ssh -c 'cd /var/www ; composer install'`
- Start testing

If you want to use docker:

- Build and run it: `docker-compose up -d --build`
- Run composer within it: `docker-compose run zf composer install`
- Start testing.

Please report any issues you notice or provide constructive feedback via comments to this pull request, or issue a pull request against my branch.

## TODO

### Installer

- [x] Prompt for caching
  - will install zend-cache
- [x] Prompt for console
  - prompt should indicate users should migrate to zf-console
  - will install zend-mvc-console
- [x] Prompt for database
  - will install zend-db
- [x] Prompt for forms
  - will install zend-form
- [x] Prompt for i18n
  - will install zend-mvc-i18n
- [x] Prompt for JSON
  - will install zend-json
- [x] Prompt for logging
  - will install zend-log
- [x] Prompt for PSR-7 middleware dispatcher
  - will install zend-psr7bridge
- [x] Prompt for sessions
  - will install zend-session
- [x] Prompt for testing facilities
  - will install zend-test *as dev requirement*
- [x] Prompt for plugins
  - will install zend-mvc-plugin-*
- [x] Prompt for zend-di integration
  - will install zend-servicemanager-di

In each case, this should register the module(s) with the application, so that the user does not need to be prompted twice.

### Documentation

- ModuleRouteListener is removed from the skeleton. This won't affect existing users, but *will* affect experienced users who originally relied on it being active in new skeleton projects.
- The `/[:controller][/:action]]` route was removed from the skeleton. Again, it will not affect existing users, but *will* affect experienced users who originally relied on it being active in new skeleton projects.

These items have been added to the [maintainers wiki](https://github.com/zendframework/maintainers/wiki/Tutorials-and-documentation-tasks-for-v3).


## Updates

- *2016-05-17: added TODO list tracking installer and documentation tasks remaining.*

Close #293
